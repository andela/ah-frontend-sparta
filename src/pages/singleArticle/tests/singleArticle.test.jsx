import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import store from '../../../store';
import { SingleArticles } from '../index';

describe('Single Article page', () => {
    const props = {
        articles: [],
        getArticle: jest.fn(),
        match: {
            params: {
                slug: 'new-slug',
            }
        }
        
    };
    let instance;
    let wrapper;
    beforeEach(() => {
        wrapper = mount(
            <Provider store={store}>
                <SingleArticles {...props} />
            </Provider>,
        );
    });

    it('should render without crashing', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('componentDidMount Method Called', () => {
        const spy = jest.spyOn(SingleArticles.prototype, 'componentDidMount');
        mount(<SingleArticles {...props} />);
        expect(spy).toHaveBeenCalledTimes(1);
    });
});
