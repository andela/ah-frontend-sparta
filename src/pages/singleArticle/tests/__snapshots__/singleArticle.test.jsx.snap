// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Article page should render without crashing 1`] = `ReactWrapper {}`;
